src/client.js

```diff
@@ -136,14 +136,29 @@ export default class StitchClient {
    */
   authenticate(providerType, options = {}) {
     // reuse existing auth if present
-    return this.auth.getAccessToken().then(accessToken => {
+    return Promise.all(
+      [
+        this.auth.getAccessToken(),
```

Comment:
[nit] It seems like it would be better to use just `getLoggedInProviderType` here right? You're only acting upon info from `providerType`

Reply:
Now that I think about it, it seems like we actually want to check `isAuthenticated` and then `getLoggedInProviderType`. The thing to keep in mind is that we only want to call `getLoggedInProviderType` if it's going to not reject or anything. Otherwise we drop out of the next then and never authenticate.

Reply:
`isAuthenticated` doesn't exist exactly– it would be a mix of checking these fields anyway.

Reply:
I think it should exist. Users need to have a decent way of checking if they are authed before doing anything. authedId seems roundabout

Reply:
Also, consider someone with existing local storage or a new SDK.  Won't `authenticate` ultimately reject because they have nothing set.